src/code/dialogs/checklist.js

```diff
@@ -119,8 +119,11 @@ const OperationChecklistDialog = WDialog.extend({
           oif.value = value;
           oif.size = 3;
           oif.disabled = !canWrite;
+          oif.type = "number";
+          oif.step = 1;
           L.DomEvent.on(oif, "change", (ev) => {
             L.DomEvent.stop(ev);
+            oif.value = parseInt(oif.value, 10);
```

Comment:
You can leave this part to the `model/task.ts` file, using `Math.trunc(o) || 0` (instead of the `+o || 0` I did)